src/prefect/blocks/core.py

```diff
@@ -629,7 +629,11 @@ def _generate_code_example(cls) -> str:
         """Generates a default code example for the current class"""
         qualified_name = to_qualified_name(cls)
         module_str = ".".join(qualified_name.split(".")[:-1])
-        class_name = cls.__name__
+        class_name = (
+            cls.__pydantic_generic_metadata__.get("origin")
+            if hasattr(cls, "__pydantic_generic_metadata__")
+            else cls.__name__
+        )
```

Comment:
```suggestion
        class_name = (
            cls.__pydantic_generic_metadata__.get("origin").__name__
            if hasattr(cls, "__pydantic_generic_metadata__")
            else cls.__name__
        )
```

Reply:
I suspect that the structure is always there and the value can be None, though.

Given that this is a class method on a class that inherits from pydantic BaseMethod might it be that the type checker already knows about `__pydantic_generic_metadata__`? I suspect that you don't need to use hasattr here and you can instead use:

```
origin = cls. __pydantic_generic_metadata__["origin"] or cls
class_name = origin.__name__
```

Reply:
good point! updated